File: app/src/main/java/com/example/test/data/users/UserRepository.java
package com.example.test.data.users;

import androidx.paging.Pager;
import androidx.paging.PagingConfig;
import androidx.paging.PagingData;
import androidx.paging.rxjava3.PagingRx;

import com.example.test.api.ApiService;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Flowable;

public class UserRepository {

    private final ApiService service;

    @Inject
    public UserRepository(ApiService service) {
        this.service = service;
    }

    public Flowable<PagingData<UserData>> getSearchUser() {
        Pager<Integer, UserData> pager = new Pager<>(
                new PagingConfig(20, 20,false),
                () -> new UserPagingSource(service)
        );
        return PagingRx.getFlowable(pager);
    }
}

File: app/src/main/java/com/example/test/data/users/UserRepository.kt
package com.example.test.data.users

import androidx.paging.Pager
import androidx.paging.PagingConfig
import androidx.paging.PagingData
import com.example.test.api.ApiService
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class UserRepository @Inject constructor(
    private val service: ApiService
) {

    fun getSearchUser(): Flow<PagingData<UserData>> {
        return Pager(
            config = PagingConfig(
                enablePlaceholders = false,
                pageSize = 20
            ),
            pagingSourceFactory = { UsersPagingSource(service) }
        ).flow
    }
}
